infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/LogReplicationConfig.java

```diff
@@ -65,43 +77,161 @@ public class LogReplicationConfig {
-                                Set<UUID> mergeOnlyStreams, int maxNumMsgPerBatch, int maxMsgSize, int cacheSize) {
-        this(streamsToReplicate, maxNumMsgPerBatch, maxMsgSize, cacheSize);
-        this.dataStreamToTagsMap = streamingTagsMap;
-        this.mergeOnlyStreams = mergeOnlyStreams;
+    /**
+     * This method is supposed to be invoked at the STANDBY side cluster to update the streams to their
+     * tags map during snapshot sync and delta sync.
+     *
+     * @param newStreamTagsMap The newly discovered / collected stream to tags map
+     * @param refresh Boolean field to check if we need to clear the present map. Set to true in Snapshot sync.
+     */
+    public void updateDataStreamToTagsMap(Map<UUID, Set<UUID>> newStreamTagsMap, boolean refresh) {
+        if (refresh) {
+            this.dataStreamToTagsMap.clear();
+        }
+        dataStreamToTagsMap.putAll(newStreamTagsMap);
+    }
+
+    /**
+     * Update the set of stream id for replicate
+     *
+     * @param streamsToReplicate Set of stream id for replicate
+     * @param refresh            True when need to clear current set of stream id
+     */
+    private void updateStreamsToReplicate(Set<TableInfo> streamsToReplicate, boolean refresh) {
+        if (refresh) {
+            this.streamsToReplicate.clear();
+        }
+
+        for (TableInfo info : streamsToReplicate) {
+            // The name field here should be a fully qualified table name
+            if (info.hasField(TableInfo.getDescriptor().findFieldByName("name"))) {
+                this.streamsToReplicate.add(CorfuRuntime.getStreamID(info.getName()));
+            } else if (info.hasField(TableInfo.getDescriptor().findFieldByName("id"))) {
+                this.streamsToReplicate.add(UUID.fromString(info.getId()));
+            }
+        }
+    }
+
+    /**
+     * This method is invoked during snapshot and log entry sync at Standby side to update the
+     * set of stream id to replicate
+     */
+    public void updateStreamsToReplicateStandby(Set<UUID> streamsToReplicate, boolean refresh) {
+        if (refresh) {
+            this.streamsToReplicate.clear();
+        }
+
+        this.streamsToReplicate.addAll(streamsToReplicate);
+    }
+
+    /**
+     * Sync with Info table upon cluster role change (leader node acquired at Active) and
+     * before a new start of snapshot sync. We do this a bit redundantly to avoid any loss
+     * of streams to replicate
+     */
+    public void syncWithInfoTable() {
+        if (configManager == null) {
+            log.warn("configManager is null! skipping sync!");
+            return;
+        }
+
+        Set<TableInfo> fetched = this.configManager.fetchStreamsToReplicate();
+        // We should refresh the stream ids here because this method is called before snapshot
+        // sync and under leadership acquire
+        updateStreamsToReplicate(fetched, true);
```

Comment:
this will clear the existing streams to replicate set.  It means the table is no longer up-to-date.  We should also update the table.  If you think about it, the method is called 'syncWithInfoTable()' but the table itself is not considered up-to-date.